src/actions/posts.js

```diff
@@ -888,6 +890,93 @@ export function getOpenGraphMetadata(url) {
     };
 }
 
+export function getFlaggedPosts() {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const userId = getCurrentUserId(state);
+        const teamId = getCurrentTeamId(state);
+
+        dispatch({type: SearchTypes.SEARCH_POSTS_REQUEST});
+        try {
+            const data = await Client4.getFlaggedPosts(userId, '', teamId);
+            dispatch(batchActions([
+                {type: SearchTypes.SEARCH_POSTS_SUCCESS},
+                {
+                    type: SearchTypes.RECEIVED_SEARCH_TERM,
+                    data: {
+                        term: null,
+                        do_search: false,
+                        is_mention_search: false
+                    }
+                },
+                {
+                    data,
+                    type: SearchTypes.RECEIVED_FLAGGED_POSTS
+                },
+                {
+                    data,
+                    type: SearchTypes.RECEIVED_SEARCH_POSTS
+                }
+            ]));
+
+            getProfilesAndStatusesForPosts(data.posts, dispatch, getState);
+            return data;
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch);
+            dispatch(batchActions([
+                {type: SearchTypes.SEARCH_POSTS_FAILURE},
+                logError(error)(dispatch)
+            ]), getState);
+            return null;
+        }
+    };
+}
+
+export function getPinnedPosts(channelId) {
```

Comment:
Can we do the same for this action as my comments for the `getFlaggedPosts` one?